File: src/main/java/de/tectoast/emolga/commands/draft/SkipPickCommand.java
package de.tectoast.emolga.commands.draft;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.emolga.utils.draft.Draft;
import net.dv8tion.jda.api.Permission;
import net.dv8tion.jda.api.entities.TextChannel;

public class SkipPickCommand extends Command {
    public SkipPickCommand() {
        super("skippick", "Skippe eine Person beim Draft", CommandCategory.Draft);
        setCustomPermissions(m -> m.hasPermission(Permission.ADMINISTRATOR) || m.getRoles().stream().anyMatch(r -> r.getId().equals("702233714360582154")));
        setArgumentTemplate(ArgumentManagerTemplate.noArgs());
    }

    @Override
    public void process(GuildCommandEvent e) {
        TextChannel tc = e.getChannel();
        Draft d = Draft.getDraftByChannel(tc);
        if (d != null) {
            d.timer(Draft.TimerReason.SKIP);
        }
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/draft/SkipPickCommand.kt
package de.tectoast.emolga.commands.draft

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.emolga.database.exposed.DraftAdmins
import de.tectoast.emolga.utils.json.emolga.draft.League
import net.dv8tion.jda.api.Permission

class SkipPickCommand : Command("skippick", "Skippe eine Person beim Draft", CommandCategory.Draft) {
    init {
        setCustomPermissions { mem ->
            mem.hasPermission(Permission.ADMINISTRATOR) || DraftAdmins.isAdmin(mem.guild.idLong, mem)
        }
        argumentTemplate = ArgumentManagerTemplate.noArgs()
        slash(true, *draftGuilds)
    }

    override suspend fun process(e: GuildCommandEvent) {
        val tc = e.textChannel
        val d = League.onlyChannel(tc.idLong) ?: return e.reply(
            "Es läuft zurzeit kein Draft in diesem Channel!",
            ephemeral = true
        )
        d.triggerTimer(League.TimerReason.SKIP, e.member.idLong)
        e.reply("+1", ephemeral = true)
    }
}
